vpn-internal/src/main/java/com/duckduckgo/vpn/internal/feature/transparency/TransparencyTrackerDetectorInterceptor.kt

```diff
@@ -37,12 +40,25 @@ class TransparencyTrackerDetectorInterceptor @Inject constructor() : VpnTrackerD
         hostname: String,
         packageId: String
     ): RequestTrackerType? {
-        return if (enable.get()) {
-            Timber.v("Transparency mode: Not Tracker returned for $packageId / $hostname")
-            RequestTrackerType.NotTracker(hostname)
-        } else {
-            Timber.v("Transparency mode: Not intercepting for $packageId / $hostname")
-            null
+        return when {
+            transparencyModeBugFixForAndroid12(packageId) -> {
+                Timber.v("Transparency mode for Android 12 and DDG: Not Tracker returned for $packageId / $hostname")
+                RequestTrackerType.NotTracker(hostname)
+            }
+            enable.get() -> {
+                Timber.v("Transparency mode: Not Tracker returned for $packageId / $hostname")
+                RequestTrackerType.NotTracker(hostname)
+            }
+            else -> {
+                Timber.v("Transparency mode: Not intercepting for $packageId / $hostname")
+                null
+            }
         }
     }
+
+    // https://issuetracker.google.com/issues/217570500
+    // https://app.asana.com/0/1174433894299346/1201657419006650
+    private fun transparencyModeBugFixForAndroid12(packageId: String): Boolean {
+        return appBuildConfig.sdkInt > Build.VERSION_CODES.S && VpnExclusionList.isDdgApp(packageId)
```

Comment:
We need `>= S` otherwise this will not work

```suggestion
        return appBuildConfig.sdkInt >= Build.VERSION_CODES.S && VpnExclusionList.isDdgApp(packageId)
```